File: backend/models/store/store.go
package store

import (
	"time"

	"github.com/google/uuid"
)

type Store struct {
	ID              uuid.UUID `json:"id" gorm:"type:uuid;primaryKey;default:gen_random_uuid()"`
	UserID          uuid.UUID `json:"user_id" gorm:"type:uuid;not null"`
	StoreTemplateID uuid.UUID `json:"store_template_id" gorm:"type:uuid"`
	Name            string    `json:"name" gorm:"not null"`
	Picture         string    `json:"picture"`
	Type            string    `json:"type" `
	Description     string    `json:"description"`
	IsActive        bool      `json:"is_active" gorm:"default:true"`
	CreatedAt       time.Time `json:"created_at"`
	UpdatedAt       time.Time `json:"updated_at"`
}

func (Store) TableName() string {
	return "stores"
}

File: frontend/types/store.ts
export interface Store {
  id: string;
  user_id: string;
  store_template_id: string;
  name: string;
  picture: string;
  type: string;
  description: string;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface MultiStoreResp {
  stores?: Store[];
  error?: string;
}

export interface SingleStoreResp {
  store?: Store;
  error?: string;
}